### DISCUSSION TITLE
rendering pygame

### DISCUSSION THREAD

**Discussion by luigiberducci** [f1tenth/f1tenth_gym#67] [Show and tell]
Hi guys,
I set up a basic implementation of the rendering in pygame in the branch `dev_pygame`

Something that is missing is:
- the track is vertically flipped: the poses must be fixed
- there is no zoom in: but it would be nice to center the window on the ego and only see the surroundings
- improve the speed: at the moment, I use skip frame to render at 30 fps because matching the sim frequency is too slow

If any of you want to continue, we can coordinate in this thread!
Luigi

**Comment by naman-gupta99** [comment]
Hey! I'd be interested in working on this. Though, I want to know what is the benfit in using pygame?

**Reply by luigiberducci** [reply]
Hi, thanks for your message! 
The benefit of using pygame is that would make our life easier in grabbing the frame and making videos out of it.
While this is common to monitor the agent behavior during training, it is currently extremely cumbersome/slow with pyglet.

**Reply by naman-gupta99** [reply]
Thanks for clearing this out.

I am really new to simulators and I really want to work on them. I would love to contribute to this issue, I would love if I could get some pointers on how to start working on this it would be great.
